scan.rb

```diff
@@ -11,13 +11,33 @@ def get_option(key)
   selected.split('=').last
 end
 
+def write_results(hostname, response, output) 
+  begin
+    puts " Start writing final results"
```

Comment:
This is executed every time a result is found. Either update the copy or remove it. Perhaps capture the results in an object and then write it in one go after it's finished.

Reply:
done